actions/shell/action.yml

```diff
@@ -0,0 +1,31 @@
+name: 'Shell Action'
+description: 'Invoke wake inside the environment-blockci-sifive container'
+inputs:
+  command:
+    description: |
+      The command which will be executed by bash inside the workspace.
+    required: true
+  workspace:
+    description: |
+      The location (relative to the build directory) of the root of the wit
+      workspace.  This should typically match wherever wit/actions/init ran.
+    required: false
+    default: .
+  version:
+    description: |
+      The version of the environment-blockci-sifive docker image to use.  By
+      default, the same version is used as the release tag of the action.
+    required: false
+    default: 0.7.0
+
+runs:
+  using: "composite"
+  steps:
+    - run: |
+        set -xeuo pipefail
+        docker run --rm \
+          --device /dev/fuse --cap-add SYS_ADMIN --security-opt apparmor:unconfined \
+          --volume="$GITHUB_WORKSPACE/${{ inputs.workspace }}:/mnt/workspace" --workdir="/mnt/workspace" \
+          sifive/environment-blockci:${{ inputs.version }} \
+          /bin/bash -c '${{ inputs.command }}'
```

Comment:
I'm worried about proper quoting and escaping. If you have single quotes inside of your string, it'll inadvertently end the string early. It's pretty common for one to make use of quoted strings in a bash command.

This might be a bit roundabout, but what if you use a heredoc and assign the value into a bash variable, and then you can use normal bash quoting rules for strings?

I'm thinking (https://stackoverflow.com/a/1655389/7433423):

```
read -r -d '' COMMAND <<'EOF'
${{ inputs.command }}
EOF

docker run --rm ... /bin/bash -c "$COMMAND"

Reply:
Couldn't we just provide stdin to the shell script?

```
docker run --rm ... /bin/bash << ENVIRONMENT_BLOCKCI_SIFIVE_SHELL_RULE${{ inputs.command }}
ENVIRONMENT_BLOCKCI_SIFIVE_SHELL_RULE
```

Reply:
That works too; I've never thought about piping a heredoc to bash, but I think that'd work.

Reply:
This does not appear to have worked.

Reply:
I've solved this using an env: